x-pack/plugins/global_search/server/services/context.test.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { httpServerMock, coreMock } from '../../../../../src/core/server/mocks';
+import { getContextFactory } from './context';
+
+describe('getContextFactory', () => {
+  it('returns a GlobalSearchProviderContext bound to the request', () => {
+    const coreStart = coreMock.createStart();
+    const request = httpServerMock.createKibanaRequest();
+
+    const factory = getContextFactory(coreStart);
+    const context = factory(request);
+
+    expect(coreStart.savedObjects.getScopedClient).toHaveBeenCalledTimes(1);
```

Comment:
I don't know if it's worth adding such a detailed test. The only logic in the factory that `uiSettings.client` uses the same `soClient` instance. Which can be considered an implementation detail.